pkg/apis/monitoring/v1/types.go

```diff
@@ -1626,3 +1640,58 @@ func (l *PrometheusRuleList) DeepCopyObject() runtime.Object {
 type ProbeTLSConfig struct {
 	SafeTLSConfig `json:",inline"`
 }
+
+// AuthTLSConfig specifies TLS configuration parameters.
+// +k8s:openapi-gen=true
+type AuthTLSConfig struct {
+	SafeTLSConfig `json:",inline"`
+}
+
+// AuthCredentials allows to specify authorization credentials as either
+// BearerToken, BasicAuth, OAuth or TLS certificate. Fields are mutually exclusive.
+type AuthCredentials struct {
+	// Secret to read or mount
+	BearerTokenSecret v1.SecretKeySelector `json:"bearerTokenSecret,omitempty"`
+	// BasicAuth credentials.
+	BasicAuth *BasicAuth `json:"basicAuth,omitempty"`
+	// TODO add the rest too and validate, see prom/common
+	// config/http_config.go
+	// OAuth2 credentials. Only valid in Prometheus versions 2.27.0 and newer.
+	// OAuth2 *OAuth2 `json:"oauth2,omitempty"`
+	// TLSConfig for this Authorization section
+	// TLSConfig *AuthTLSConfig `json:"authTLSConfig,omitempty"`
+}
+
+// Authorization contains optional `Authorization` header configuration.
+// This section is only understood by  versions of Prometheus >= 2.26.0.
+// +k8s:openapi-gen=true
+type Authorization struct {
+	// Set the authentication type. Defaults to Bearer
+	Type string `json:"type,omitempty"`
+	// Secret to read or mount
+	Credentials *AuthCredentials `json:"credentials,omitempty"`
+	// File to read a secret from
+	CredentialsFile string `json:"credentialsFile,omitempty"`
+}
+
+// Validate semantically validates the given Authorization section.
+func (c *Authorization) Validate() error {
+	if c.Credentials != nil {
+		if c.CredentialsFile != "" {
+			return &AuthorizationValidationError{"Authorization can not specify both Credentials and CredentialsFile"}
+		}
+	}
```

Comment:
```suggestion
	if c.Credentials != nil && c.CredentialsFile != "" {
		return &AuthorizationValidationError{"Authorization can not specify both Credentials and CredentialsFile"}
	}
```

Reply:
So far the intention is to add more validation. Lets keep it like this for now, I'll change it if no more validation code is needed.